File: MyAngular-1&WebAPI/domain/product.ts
module App.Domain {
    export interface IProduct {
        productId: number;
        productName: string;
        productCode: string;
        releaseDate: Date;
        price: number;
        cost:number;
        description: string;
        imageUrl: string;
        calculateDiscount(percent: number): number;
        tags: string[];
        
}

    export class Product implements IProduct {

        constructor(public productId: number,
            public productName: string,
            public productCode: string,
            public releaseDate: Date,
            public cost: number,
            public price: number,
            public description: string,
            public imageUrl: string,
            public tags: any[]) {
        }

        calculateDiscount(percent: number): number {
            return this.price - (this.price * percent / 100);
        }
    }
}

File: MyAngular-1&WebAPI/MyLOB.WebAPI/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MyLOB.WebAPI.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        public string Description { get; set; }
        public string ProductCode { get; set; }
        public string ProductName { get; set; }
        public decimal Price { get; set; }
        public decimal starRating { get; set; }
        
        public decimal Cost { get; set; }
        public DateTime ReleaseDate { get; set; }
        public string ImageUrl { get; set; }
        public string[] Tags{ get; set; }

        public Product()
        {
            Tags = new[] {"abc", "ad", "fg"};
        }
    }
}